pkg/controller/template/render.go

```diff
@@ -196,19 +197,30 @@ func filterTemplates(toFilter map[string]string, path string, config *RenderConf
 	return filepath.Walk(path, walkFn)
 }
 
+func getPaths(config *RenderConfig, platformString string) []string {
+	platformBasedPaths := []string{platformBase, platformString}
+	if onPremPlatform(config.Infra.Status.PlatformStatus.Type) {
+		platformBasedPaths = append(platformBasedPaths, platformOnPrem)
+	}
+
+	if config.Infra.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode {
```

Comment:
Just to be extra safe, `ControlPlaneTopology` can never be unset, right?

Reply:
Yep, as far as i know. We don't allow to go from sno to ha for sure